pexpect/pty_spawn.py

```diff
@@ -430,11 +452,7 @@ def read_nonblocking(self, size=1, timeout=-1):
         # For this case, I test isalive() before doing any reading.
         # If isalive() is false, then I pretend that this is the same as EOF.
```

Comment:
Are you suggesting that Solaris might block even when calling __select with a timeout of 0?  It looks to me like these comments must refer to an earlier version of this code where timeout may not have been 0.